components/Tabs/Tabs.js

```diff
@@ -1,63 +1,61 @@
 class TabLink {
-  constructor(tabElement){
+  constructor(tabElement) {
     // assign this.tabElement to the tabElement DOM reference
-    // this.tabElement;
-    
+    this.tabElement = tabElement;
+
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
-    
-    // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
-    
-    /* <- Delete this comment block when you work on the if statement
+    this.tabData = this.tabElement.dataset.tab;
+
+    // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:
+
     // Check to see if this.tabData is equal to 'all'
-    if(){
+    if (this.tabData === "all") {
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+      this.cards = document.querySelectorAll(".card");
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
+      this.cards = this.cards = document.querySelectorAll(
+        `.card[data-tab='${this.tabData}']`
+      );
```

Comment:
This could also be written as a ternary statement...
```js
this.cards = this.tabData === 'all'
  ? document.querySelectorAll(".card")
  : document.querySelectorAll(`.card[data-tab='${this.tabData}']`);
```